cfme/infrastructure/host.py

```diff
@@ -403,8 +364,154 @@ def _select_rows(indexes):
+        )
 
-@navigator.register(Host)
+    def wait_for_delete(self):
+        """Waits for the host to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for a host to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the host to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class HostCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.Host`."""
+
+    ENTITY = Host
+
+    def create(self, name, provider, credentials=None, hostname=None, ip_address=None,
+               host_platform=None, custom_ident=None, ipmi_address=None, mac_address=None,
+               ipmi_credentials=None, cancel=False, validate_credentials=False):
+        """Creates a host in the UI.
+
+        Args:
+           cancel (bool): Whether to cancel out of the creation. The cancel is done after all the
+               information present in the Host has been filled in the UI.
+           validate_credentials (bool): Whether to validate credentials - if True and the
+               credentials are invalid, an error will be raised.
+        """
+        view = navigate_to(self, "Add")
+        view.fill({
+            "name": name,
+            "hostname": hostname or ip_address,
+            "host_platform": host_platform,
+            "custom_ident": custom_ident,
+            "ipmi_address": ipmi_address,
+            "mac_address": mac_address
+        })
+        if credentials is not None:
+            view.endpoints.default.fill(credentials.view_value_mapping)
+            if validate_credentials:
+                view.endpoints.default.validate_button.click()
+        if ipmi_credentials is not None:
+            view.endpoints.ipmi.fill(ipmi_credentials.view_value_mapping)
+            if validate_credentials:
+                view.endpoints.ipmi.validate_button.click()
+        if not cancel:
+            view.add_button.click()
+            flash_message = 'Host / Node " {}" was added'.format(name)
+        else:
+            view.cancel_button.click()
+            flash_message = "Add of new Host / Node was cancelled by the user"
+        host = self.instantiate(name=name, hostname=hostname, ip_address=ip_address,
+                                custom_ident=custom_ident, host_platform=host_platform,
+                                ipmi_address=ipmi_address, mac_address=mac_address,
+                                credentials=credentials, ipmi_credentials=ipmi_credentials,
+                                provider=provider)
+        view = host.create_view(HostsView)
+        assert view.is_displayed
+        view.flash.assert_success_message(flash_message)
+        return host
+
+    def all(self, provider):
+        """returning all Object Store Objects"""
+        view = navigate_to(self, 'All')
+        view.toolbar.view_selector.select("Grid View")
```

Comment:
**REQUIRED**
It's not necessary, `including_entities` attribute of `HostView` represents all types of entities.